Mage.Sets/src/mage/cards/l/LukkaBoundToRuin.java

```diff
@@ -0,0 +1,160 @@
+package mage.cards.l;
+
+import mage.ConditionalMana;
+import mage.MageObject;
+import mage.Mana;
+import mage.abilities.Ability;
+import mage.abilities.LoyaltyAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.GreatestPowerAmongControlledCreaturesValue;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.DamageMultiEffect;
+import mage.abilities.keyword.CompleatedAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.permanent.token.PhyrexianBeastToxicToken;
+import mage.players.Player;
+import mage.target.common.TargetCreatureOrPlaneswalkerAmount;
+import mage.target.targetadjustment.TargetAdjuster;
+
+import java.util.UUID;
+
+/**
+ * @author miesma
+ */
+public class LukkaBoundToRuin extends CardImpl {
+
+    public LukkaBoundToRuin(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.PLANESWALKER}, "{2}{R}{R/G/P}{G}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.LUKKA);
+        this.setStartingLoyalty(5);
+
+        // Compleated
+        this.addAbility(CompleatedAbility.getInstance());
+
+        // +1: Add {R}{G}. Spend this mana only to cast creature spells or activate abilities of creatures.
+        Ability ability = new LoyaltyAbility(new LukkaBoundToRuinManaEffect(),1);
+        this.addAbility(ability);
+
+        // −1: Create a 3/3 green Phyrexian Beast creature token with toxic 1.
+        ability = new LoyaltyAbility(new CreateTokenEffect(new PhyrexianBeastToxicToken()),-1);
+        this.addAbility(ability);
+
+        // −4: Lukka deals X damage divided as you choose among any number of target creatures and/or planeswalkers,
+        // where X is the greatest power among creatures you controlled as you activated this ability.
+        DynamicValue xValue = GreatestPowerAmongControlledCreaturesValue.instance;
+        ability = new LoyaltyAbility(new LukkaBoundToRuinDamageEffect(xValue), -4);
```

Comment:
you don't need a custom damage effect here, just use `DamageMultiEffect` and override the text and the target will handle the rest

Reply:
Yes - thank you - I think I just worked myself into this as I did not quite understood what exactly the effect does with targets and choosing an unlimited amount at the time felt wrong to me